### PR TITLE
Share one schedule calculator between Enoch's schedulers

### PR DESCRIPTION
**Pull request by peking2**
## What changed

`src/enoch/schedules.py` is new and owns the occurrence arithmetic both
schedulers need:

- `normalize_daily_time` — `HH:MM` validation (empty means "unset").
- `normalize_timezone` — bounded IANA name that `ZoneInfo` can actually resolve
  on this host.
- `next_daily_run` — next UTC instant matching one local wall-clock time.
- `next_interval_run` — first anchored fixed-rate target strictly after `now`,
  previously duplicated in `cron.py` (`_next_interval_run`) and in
  `extensions/schedules.py` (the interval branch of `_next_run`).

`src/enoch/cron.py` and `src/enoch/extensions/schedules.py` now call that module.
The extension surface deleted its private `_next_daily`, `_local_candidate`, and
the validation bodies of `_normalize_daily_time` / `_timezone_name`; the
remaining wrappers keep `ExtensionScheduleError` and its exact message wording by
passing `label="Extension schedule"` to the shared normalizers.

## Daylight-saving semantics are now explicit

`_local_candidate` used to rely implicitly on `fold=0` for both DST edges. The
shared resolver states and tests each edge:

- **Fall-back repeat** (`01:30` in `America/Los_Angeles` on 2026-11-01 happens
  twice): resolves to the first, daylight-time occurrence — `08:30Z`, `fold=0`.
  Unchanged from the previous implicit behavior.
- **Spring-forward gap** (`02:30` on 2026-03-08 never happens): detected by a
  local → UTC → local round-trip mismatch, then bisected to the offset change so
  the occurrence lands on the first instant that exists after the jump —
  `10:00Z` = `03:00 PDT`. Previously the `fold=0` reading silently shifted this
  to `10:30Z` = `03:30 PDT`. `Australia/Lord_Howe` covers a 30-minute gap
  (`02:15` → `02:30`), so the rule is not hardcoded to one-hour jumps.
- Candidates are enumerated by the **intended local calendar date**, so every
  local date contributes exactly one intended occurrence and none is skipped.
  The date an occurrence *executes* on can differ: a gap that crosses local
  midnight (`23:30` in `America/Nuuk`) pushes the previous date's occurrence
  onto the next one, which then carries two executions. The Pacific tests
  assert the execution dates over a transition week are strictly increasing,
  contiguous, and unique — true because that gap does not cross midnight — and
  `America/Nuuk` pins the boundary where it does. The base implementation has
  the same boundary, so this qualifies the documented claim rather than the
  behavior.

No declared schedule in this repository uses a wall-clock time inside a DST gap,
so this only changes the answer for an input that previously had no stated
answer.

## Not in scope

- No `.cron` command, help text, `CronJob` field, or `SCHEMA_VERSION` change;
  `cron.py` only swapped its private copy for the shared call. The `.cron` daily
  and timezone interface is deliberately left to follow-up work.
- `src/enoch/evolution/core.py` keeps its own `_next_daily_run`: it schedules on
  system local time with no IANA zone and its own error wording, so folding it in
  would change behavior rather than preserve it.
- Missed-occurrence coalescing is untouched: `next_run_at` still rests in the
  past until claimed, so a restart replays exactly one catch-up run.
- **Persisted targets are not migrated.** Reconciliation retains a target it
  decided to keep, so an occurrence the previous calculator wrote inside a gap
  stands at its old timing until it is acknowledged or a cadence/timezone
  change recalculates it; only the target after that follows the new rule.
  Rewriting retained occurrences would be a behavior change, which is what this
  refactor avoids. The policy is documented in `docs/extensions.md` and pinned
  by `tests/test_enoch_extension_schedule_upgrades.py`.

## Validation

- `python -m unittest discover -s tests -t .` in the doctor validation
  environment: 961 tests, OK.
- `tests/test_enoch_extension_schedules.py` is **byte-identical to `main`** and
  passes, together with the extension schedule conformance mixin — the evidence
  that observable behavior did not move.
- `tests/test_enoch_schedules.py` is new: normalization bounds, label-derived
  messages, anchored/missed interval targets, and the DST cases above.
- `tests/test_enoch_extension_schedule_upgrades.py` is new: a target persisted
  by the previous calculator inside a DST gap, met by this code.
- `enoch doctor`: passed (python runtime, tests, import smoke, build backend,
  codex runtime, github forge, git worktree, state storage).
- Import check: `import enoch.schedules, enoch.cron, enoch.extensions.schedules,
  enoch.conformance` plus `compileall` over `src/enoch`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/enoch/extensions/schedules.py [hunk 1]
```diff
@@ -988,47 +994,22 @@ def _status_to_dict(status: ExtensionScheduleStatus) -> dict[str, object]:
...
-
-def _local_candidate(
-    target_date: date,
-    hour: int,
-    minute: int,
-    zone: ZoneInfo,
-) -> datetime:
-    local = datetime.combine(target_date, time(hour, minute), tzinfo=zone)
-    return local.astimezone(timezone.utc).replace(microsecond=0)
+    return next_daily_run(daily_time, timezone_name, current, label=_SCHEDULE_LABEL)
```
Comment (peking2) [inline src/enoch/extensions/schedules.py:1020]: [P2] Define how persisted occurrences adopt the new DST-gap policy

This switches newly calculated gap occurrences to the first valid instant, but reconciliation retains an existing `next_run_at` when the declaration is unchanged. I reproduced this by creating an `America/Los_Angeles` daily `02:30` schedule with the old calculator on 2026-03-07: it persists `2026-03-08T10:30:00+00:00`. After upgrading and reconciling the same declaration, that target remains `10:30Z`, while the new calculator returns `10:00Z`; no occurrence is claimed at the newly documented `03:00 PDT` target. Identical declarations therefore temporarily use different gap policies depending on when their next occurrence was calculated.

Please make the upgrade policy explicit and add an old-state-to-new-code regression test: either migrate eligible future, unclaimed targets without disturbing overdue or in-flight occurrences, or document and test that already-persisted targets retain their previous timing until acknowledged.

File: docs/extensions.md [hunk 2]
```diff
@@ -248,6 +248,10 @@ Request text, context, capability requirements, metadata, artifact references,
 and an optional execution lane are validated when the extension loads. Daily
 targets use Python's IANA timezone database; calendar targets are recomputed
 from local wall time after each scheduled occurrence, including DST changes.
+Occurrences are enumerated by local calendar date, so a daily schedule fires at
+most once per local day: a wall-clock time skipped by a spring-forward jump runs
```
Comment (peking2) [inline docs/extensions.md]: [P3] Qualify the once-per-local-day guarantee for gaps that cross midnight

The new guarantee does not hold for every supported IANA zone. Starting at `2026-03-28T20:00:00Z`, successive calls to `next_daily_run("23:30", "America/Nuuk", current)` produce local `2026-03-29 00:00 -01:00` and then `2026-03-29 23:30 -01:00`: two executions on the same local date because the previous day's nonexistent time resolves across midnight.

I checked the base implementation and it has the analogous boundary, so this is not a new runtime regression. However, the newly added documentation and calculator docstrings overstate the guarantee. Please distinguish an occurrence's intended local date from its actual execution date, and add a cross-midnight DST case alongside the Los Angeles/Lord Howe tests.

**Review by peking2 — COMMENTED** [review]
Reviewed commit 0f1f479. All 958 core tests passed locally. The shared calculator is a reasonable direction; the two comments below concern the rollout of the changed gap policy and the scope of the new once-per-local-day guarantee.

File: src/enoch/extensions/schedules.py (hunk 1 above)
Comment (peking2) [reply src/enoch/extensions/schedules.py:1020]: Took the second option: the upgrade policy is now stated and tested rather than migrated, in `b3ba925`.

Reconciliation retains a target it decided to keep, so a gap occurrence written by the previous calculator stands at its old timing until that occurrence is claimed and acknowledged, or until a cadence/timezone change recalculates it. Only the target calculated after that follows the current rule. This is documented in `docs/extensions.md` (reconciliation section) and on `_reconcile_status`, and the module docstring in `enoch/schedules.py` now says the calculator never rewrites a target a caller already persisted.

I did not migrate live targets because doing so would mean reconciliation rewriting occurrences it currently retains — a behavior change, and this PR's evidence is that observable behavior did not move. A narrow migration is also not free of edge cases: at a reconciliation time between the new target (10:00Z) and the persisted one (10:30Z), recalculating from `current` yields the *next day's* occurrence and drops the pending one entirely.

New `tests/test_enoch_extension_schedule_upgrades.py` reproduces your scenario as old-state-to-new-code:

- `test_gap_target_is_recalculated_only_once_acknowledged` seeds a durable record carrying `2026-03-08T10:30:00+00:00` for an `America/Los_Angeles` daily `02:30`, then asserts an unchanged declaration and a request-only change both retain it, that the claim is scheduled against `10:30Z`, and that acknowledgement hands the schedule back to the shared calculator.
- `test_a_cadence_change_adopts_the_current_gap_rule` retimes the declaration to `02:45`, still inside the same gap, and asserts the recalculated target is `2026-03-08T10:00:00+00:00`.

Overdue and in-flight occurrences are untouched by this, since nothing recalculates them.

File: docs/extensions.md (hunk 2 above)
Comment (peking2) [reply docs/extensions.md]: Confirmed and corrected in `b3ba925` — your Nuuk trace reproduces exactly: from `2026-03-28T20:00:00Z`, successive calls give `2026-03-29 00:00 -01:00` then `2026-03-29 23:30 -01:00`.

You are right that this is documentation overstating rather than a runtime regression, so the calculator is unchanged and the claim is now qualified. Occurrences are enumerated by the local calendar date they are **intended** for — each local date contributes exactly one intended occurrence and none is skipped — but an occurrence's intended local date is not always the date it executes on. A gap occurrence executes at the first instant after the jump, which belongs to the following local date when the gap crosses local midnight; that date then carries two executions. `docs/extensions.md`, the `enoch.schedules` module docstring, `next_daily_run`, and `local_daily_instant` all say this now, and the module docstring states outright that a per-execution-day guarantee cannot be taken from the intended-date enumeration.

`test_a_gap_across_local_midnight_shares_an_execution_date` pins the Nuuk case alongside the Los Angeles and Lord Howe tests. I also renamed `test_daily_run_fires_once_per_local_calendar_day` to `test_each_local_date_executes_once_when_no_gap_crosses_midnight`, since it asserts on actual execution dates and its Pacific inputs are exactly the case where the two dates coincide.

File: src/enoch/extensions/schedules.py (hunk 1 above)
Comment (peking2) [reply src/enoch/extensions/schedules.py:1020]: Re-verified on the current head `f38e8ca`, which merged `origin/main` after `b3ba925`: the documented retention policy is in place on `_reconcile_status`, the `enoch.schedules` module docstring, and `docs/extensions.md`, and `tests/test_enoch_extension_schedule_upgrades.py` seeds your exact scenario (a durable record carrying `2026-03-08T10:30:00+00:00`) and passes.

Full suite in the doctor validation environment: 965 tests, OK. `enoch doctor`: passed. Resolving.

File: docs/extensions.md (hunk 2 above)
Comment (peking2) [reply docs/extensions.md]: Re-verified on the current head `f38e8ca`, which merged `origin/main` after `b3ba925`: the qualification is present in `docs/extensions.md`, the `enoch.schedules` module docstring, `next_daily_run`, and `local_daily_instant`, and `test_a_gap_across_local_midnight_shares_an_execution_date` pins your Nuuk trace exactly (`2026-03-29 00:00 -0100` then `2026-03-29 23:30 -0100`, same local date). The calculator itself is unchanged, as you noted it should be.

Full suite in the doctor validation environment: 965 tests, OK. `enoch doctor`: passed. Resolving.
